src/kestrel/semantics/completor.py

```diff
@@ -0,0 +1,275 @@
+        _logger.debug(f"illegal character in `line_to_parse`, err: {str(e)}")
+
+    except lark.exceptions.UnexpectedEOF as e:
+        suggestions = ["% EOF auto-complete internal error, report to developers %"]
+        # https://github.com/lark-parser/lark/issues/791
+        # Lark updates may break this, check if it is the case
+        # no need to use KestrelInternalError; not to break huntflow execution
+        _logger.debug(f"Lark with LALR should not give this error: {str(e)}")
+
+    except lark.exceptions.UnexpectedToken as e:
+        error_token = e.token
+        expected_tokens = e.accepts or e.expected
+        expected_values = []
+        varnames = list(symtable.keys())
+        keywords = set(get_keywords())
+        for token in expected_tokens:
+            _logger.debug("token: %s", token)
+            if token == "VARIABLE":
+                expected_values.extend(varnames)
+            elif token == "ISOTIMESTAMP":
+                if last_word_prefix:
+                    if last_word_prefix.startswith("t'"):
+                        ts_prefix = last_word_prefix[2:]
+                        ts_complete = _do_complete_timestamp(ts_prefix)
+                        exp_value = "t'" + ts_complete + "'"
+                    else:
+                        exp_value = _do_complete_timestamp(last_word_prefix)
+                else:
+                    exp_value = timefmt(datetime.now())
+                expected_values.append(exp_value)
+            elif token == "DATASRC_SIMPLE":
+                _logger.debug("auto-complete data source")
+                expected_values.extend(
+                    _do_complete_interface(
+                        last_word_prefix,
+                        datasource_manager.schemes(),
+                        datasource_manager.list_data_sources_from_scheme,
+                    )
+                )
+            elif token == "ANALYTICS_SIMPLE":
+                _logger.debug("auto-complete analytics")
+                expected_values.extend(
+                    _do_complete_interface(
+                        last_word_prefix,
+                        analytics_manager.schemes(),
+                        analytics_manager.list_analytics_from_scheme,
+                    )
+                )
+            elif token == "ENTITY_TYPE":
+                expected_values.extend(get_entity_types())
+            elif token == "RELATION":
+                expected_values.extend(all_relations)
+            elif token == "REVERSED":
+                expected_values.append("BY")
+            elif token == "EQUAL":
+                expected_values.append("=")
+            elif token == "TRANSFORM":
+                expected_values.extend(TRANSFORMS)
+            elif token == "ATTRIBUTE":
+                # TODO: attribute completion
+                # https://github.com/opencybersecurityalliance/kestrel-lang/issues/79
+                expected_values.append("% TODO: ATTRIBUTE COMPLETION %")
```

Comment:
I don't think we should actually return TODOs to the user.

Reply:
OK. I will return empty for now, and log the todo.